src/test/javascript/spec/component/team/teams-import-dialog.component.spec.ts

```diff
@@ -0,0 +1,735 @@
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { DebugElement } from '@angular/core';
+import { ComponentFixture, fakeAsync, TestBed, tick, waitForAsync } from '@angular/core/testing';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { User } from 'app/core/user/user.model';
+import { Exercise } from 'app/entities/exercise.model';
+import { Team, TeamImportStrategyType } from 'app/entities/team.model';
+import { ArtemisTeamModule } from 'app/exercises/shared/team/team.module';
+import { TeamService } from 'app/exercises/shared/team/team.service';
+import { TeamsImportDialogComponent } from 'app/exercises/shared/team/teams-import-dialog/teams-import-dialog.component';
+import * as chai from 'chai';
+import { flatMap } from 'lodash';
+import { JhiAlertService, JhiEventManager } from 'ng-jhipster';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { of, throwError } from 'rxjs';
+import { SinonSpy, SinonStub, spy, stub } from 'sinon';
+import * as sinonChai from 'sinon-chai';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+import {
+    mockExercise,
+    mockSourceExercise,
+    mockSourceTeams,
+    mockSourceTeamStudents,
+    mockTeam,
+    mockTeams,
+    MockTeamService,
+    mockTeamStudents,
+} from '../../helpers/mocks/service/mock-team.service';
+import { ArtemisTestModule } from '../../test.module';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+export const config = {
+    imports: [ArtemisTestModule, ArtemisTeamModule],
```

Comment:
This makes the test incredibly brittle and the test dependency incredible huge. use https://www.npmjs.com/package/ng-mocks to mock everything you need for artemis test module.